pkg/cvo/upgradeable.go

```diff
@@ -65,10 +65,11 @@ func defaultUpgradeableCheckIntervals() upgradeableCheckIntervals {
 func (optr *Operator) syncUpgradeable(cv *configv1.ClusterVersion) error {
 	if u := optr.getUpgradeable(); u != nil {
 		if u.RecentlyChanged(optr.upgradeableCheckIntervals.throttlePeriod(cv)) {
-			klog.V(2).Infof("Upgradeable conditions were recently checked, will try later.")
+			klog.V(2).Infof("Upgradeable conditions were recently (%s) checked, will try later.", u.At)
```

Comment:
"will try later" is ambiguous, and we have the throttle period above.  How about something like:

```go
throttlePeriod := optr.upgradeableCheckIntervals.throttlePeriod(cv)
if u.RecentlyChanged(throttlePeriod) {
  klog.V(2).Infof("Upgradeable conditions were recently (%s) checked, will try after %s.", u.At, u.At.Add(throttlePeriod))
  return nil
}
```

Or, for a larger reroll to reduce the amount of assumptions about how `RecentlyChanged` is implemented, we could do:

```go
if err := u.TimeToUpdate(optr.upgradeableCheckIntervals.throttlePeriod(cv)); err != nil {
  klog.V(2).Infof("Upgradeable conditions not ready for update: %v", err)
  return nil
}
```

and port `RecentlyChanged` returning a `bool` to `TimeToUpdate` (or some such) returning an `error` that is `nil` when it is time to update, and explains why it is not yet time to update (and possibly when we expect it to become time to update) when we want to wait longer.

Reply:
:thinking:

Reply:
I did not go for the helper returning `error` but I refactored the code in the direction that you suggested.